File: cha/bj/2436.js
/* eslint-disable radix */
const readline = require('readline');

const GCD = (a, b) => {
  if (b === 0) return a;
  return GCD(b, a % b);
};

const getCD = (num) => {
  const sqrt = Math.sqrt(num);
  const arr = [];
  for (let i = 1; i <= sqrt; i += 1) {
    if (num % i === 0) {
      if (GCD(num / i, i) === 1) {
        arr.push([i, num / i]);
      }
    }
  }
  return arr;
};

const getMinSum = (commonDevisor) => {
  let minSum = 1e9;
  let ansA;
  let ansB;
  commonDevisor.map(([x, y]) => {
    if (x + y < minSum) {
      minSum = x + y;
      ansA = x;
      ansB = y;
    }
  });
  return [ansA, ansB];
};

const solve = (gcd, lcm) => {
  const ab = parseInt(lcm / gcd);
  const commonDevisor = getCD(ab);
  const [a, b] = getMinSum(commonDevisor);
  return [gcd * a, gcd * b];
};

const input = () => {
  let a;
  let b;
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });
  rl.on('line', (line) => {
    if (!a) {
      [a, b] = line.split(' ').map((el) => +el);
      const ans = solve(a, b);
      console.log(ans.join(' '));
      rl.close();
    }
  }).on('close', () => process.exit());
};

input();

File: yang/20210329/2436.py
# import sys
# n,m=map(int,sys.stdin.readline().split())
# if m%n!=0 or n==0 or m==0:
#     exit()


# if n==m:
#     a=n
#     b=n
# else:
#     a=int(m/n)
#     k=a/n
#     b=int(m/k)

# if m!=(a//n)*(b//n)*n:
#     exit()

# if a>b:
#     temp=a
#     a=b
#     b=temp

# print(a,b)

def gcd(a,b):
    while b:
        mod = b
        b=a%b
        a=mod
    return a

g, l = map(int,input().split())
div = l//g
a,b=1,div
for i in range(1,div//2+1):
    if div%i ==0:
        c=i
        d=div//i
        if gcd(c,d) != 1:
            continue
        if a+b>c+d:
            a=c
            b=d
print(a*g,b*g)